### PR TITLE
fix: crytic issues

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

File: contracts/child/ChildToken/ChildERC20.sol [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
     constructor(
-        string memory name,
-        string memory symbol,
-        uint8 _decimals
-    ) public ERC20(name, symbol) NetworkAgnostic(name, "1", 3) {
-        _setupDecimals(_decimals);
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_
+    ) public ERC20(name_, symbol_) NetworkAgnostic(name_, ERC712_VERSION, ROOT_CHAIN_ID) {
```
Comment (arthcp) [inline contracts/child/ChildToken/ChildERC20.sol:17]: using `ERC712_VERSION` for all child tokens. Is this intended?

File: contracts/root/RootChainManager/RootChainManager.sol [hunk 2]
```diff
@@ -75,6 +75,7 @@ contract RootChainManager is IRootChainManager, Initializable, AccessControl {
         external
         only(DEFAULT_ADMIN_ROLE)
     {
+        require(newChildChainManager != address(0x0), "New child manager should not be zero");
```
Comment (arthcp) [inline contracts/root/RootChainManager/RootChainManager.sol:78]: Revert message can be changed to match the format of other ones in this contract.

File: contracts/child/ChildToken/ChildERC20.sol (hunk 1 above)
Comment (jdkanani) [reply contracts/child/ChildToken/ChildERC20.sol:17]: Yes. It was "1" hard coded before.

File: contracts/root/RootChainManager/RootChainManager.sol (hunk 2 above)
Comment (jdkanani) [reply contracts/root/RootChainManager/RootChainManager.sol]: Fixed.

File: contracts/child/ChildToken/ChildERC20.sol (hunk 1 above)
Comment (arthcp) [reply contracts/child/ChildToken/ChildERC20.sol:17]: but why name it `ERC712_VERSION` and not `TOKEN_VERSION`?

Comment (arthcp) [reply contracts/child/ChildToken/ChildERC20.sol:17]: nevermind I kept seeing 712 as 721 :P

File: contracts/lib/MerklePatriciaProof.sol [hunk 3]
```diff
@@ -102,14 +102,16 @@ library MerklePatriciaProof {
...
+        return false;
     }
 
     function _nibblesToTraverse(
         bytes memory encodedPartialPath,
         bytes memory path,
         uint256 pathPtr
     ) private pure returns (uint256) {
-        uint256 len;
+        uint256 len = 0;
```
Comment (imyourm8) [inline contracts/lib/MerklePatriciaProof.sol:114]: this is totally unnecessary

File: contracts/lib/MerklePatriciaProof.sol [hunk 4]
```diff
@@ -136,7 +138,7 @@ library MerklePatriciaProof {
         pure
         returns (bytes memory)
     {
-        bytes memory nibbles;
+        bytes memory nibbles = "";
```
Comment (imyourm8) [inline contracts/lib/MerklePatriciaProof.sol:141]: and this.

File: contracts/lib/MerklePatriciaProof.sol (hunk 3 above)
Comment (jdkanani) [reply contracts/lib/MerklePatriciaProof.sol:114]: Just to make crytic happy.

File: contracts/lib/MerklePatriciaProof.sol (hunk 4 above)
Comment (jdkanani) [reply contracts/lib/MerklePatriciaProof.sol:141]: Just to make crytic happy :)

**Pull request merged**